### PR TITLE
Editing add skips

### PR DESCRIPTION
**Pull request by gwenllian1**
If a move is edited, all future moves that are made illegal are replaced with skips
Kat + Llio.

https://user-images.githubusercontent.com/64511606/190897721-dab5e10d-c5a7-4138-af3c-c5c0c2d959be.mp4

### CODE REVIEW

File: packages/TorneloScoresheet/src/hooks/appMode/editMoveState.ts [hunk 1]
```diff
@@ -96,15 +97,22 @@ const rebuildHistory = (
...
   // if the starting fen is null (the move was impossible) we add null to to newHistory[]
   // otherwise we add the ply with the updated starting fen to newHistory[]
   const newHistory = futureMoves.map(move => {
+    if (previousStartingFen === '') {
+      previousPly = {
+        moveNo: move.moveNo,
+        startingFen: previousPly.startingFen,
+        drawOffer: move.drawOffer,
+        player: move.player,
+        type: PlyTypes.SkipPly,
```
Comment (nocommentcode) [inline packages/TorneloScoresheet/src/hooks/appMode/editMoveState.ts:112]: Missing the legality check here, it should be performed with the fen after this skip ply is recorded

File: packages/TorneloScoresheet/src/hooks/appMode/editMoveState.ts [hunk 2]
```diff
@@ -96,15 +97,22 @@ const rebuildHistory = (
...
+      };
+      return previousPly;
```
Comment (nocommentcode) [inline packages/TorneloScoresheet/src/hooks/appMode/editMoveState.ts:114]: Also we need to update the posistion occureneces with this new fen before returning previous ply

File: packages/TorneloScoresheet/src/hooks/appMode/editMoveState.ts [hunk 3]
```diff
@@ -118,8 +126,21 @@ const rebuildHistory = (
...
+      // return a skip if move impossible
       if (!nextMoveResult) {
-        return null;
+        previousPly = {
+          moveNo: move.moveNo,
+          startingFen: previousStartingFen,
+          drawOffer: move.drawOffer,
+          player: move.player,
+          type: PlyTypes.SkipPly,
+          legality: checkMoveLegality(previousStartingFen, positionOccurances),
```
Comment (nocommentcode) [inline packages/TorneloScoresheet/src/hooks/appMode/editMoveState.ts:138]: The legality of this move needs to be checked with the fen resulting from this move, ie you need `const nextFen = chessEngine.skipTurn(previousStartingFen)` before declaring previousPly, then you need to call checkMoveLegality with this next fen,
 I just noticed there is a bug in the previous version, in line 138, it should be checking the legality with the next fen too!

File: packages/TorneloScoresheet/src/hooks/appMode/editMoveState.ts (hunk 2 above)
Comment (nocommentcode) [reply packages/TorneloScoresheet/src/hooks/appMode/editMoveState.ts:114]: and then we need to set the previous starting fen to be this new fen

**Review by nocommentcode — CHANGES_REQUESTED** [review]
Looks good, just needs a minor fix with the legality checking.
I also realised there was a bug with the previous version, I was checking the legality with the previous fen instead of the next fen for skip plys, so this needs to be changed!

**Review by nocommentcode — APPROVED** [review]
nicely done :)

**Review by nocommentcode — APPROVED** [review]
!

**Review by ksov0001 — APPROVED** [review]
Re-approving after rebase

**Pull request merged**